### DISCUSSION TITLE
npm: CALL "C:\nodejs\\node.exe" "C:\nodejs\\node_modules\npm\bin\npm-cli.js" prefix -g' is not recognized as an internal or external command

### DISCUSSION THREAD

**Discussion by jpjsm** [nodejs/help#4631] [Q&A]
### Version

node@v16.20.0

### Platform

Microsoft Windows NT 10.0.19042.0 x64

### Subsystem

npm

### What steps will reproduce the bug?

I have tried with two different versions of node: 18.15.0 and 16.20.0 with the exact same results.

Executing any npm command always throws the same error: `'CALL "C:\nodejs\\node.exe" "C:\nodejs\\node_modules\npm\bin\npm-cli.js" prefix -g' is not recognized as an internal or external command, operable program or batch file.`

### How often does it reproduce? Is there a required condition?

always

### What is the expected behavior? Why is that the expected behavior?

No error at all.
(I suppose there's no need to explain why an errorless behavior is expected)

### What do you see instead?

This is one example when using`npm`, any usage of `npm` throws the same error:
```
C:\Users\juan_pablo_jofre>npm --verbose --version
'CALL "C:\nodejs\\node.exe" "C:\nodejs\\node_modules\npm\bin\npm-cli.js" prefix -g' is not recognized as an internal or external command,
operable program or batch file.
npm verb cli C:\nodejs\node.exe C:\nodejs\node_modules\npm\bin\npm-cli.js
npm info using npm@8.19.4
npm info using node@v16.20.0
npm timing npm:load:whichnode Completed in 1ms
npm timing config:load:defaults Completed in 2ms
npm timing config:load:file:C:\nodejs\node_modules\npm\npmrc Completed in 3ms
npm timing config:load:builtin Completed in 3ms
npm timing config:load:cli Completed in 1ms
npm timing config:load:env Completed in 0ms
npm timing config:load:project Completed in 2ms
npm timing config:load:file:C:\Users\juan_pablo_jofre\.npmrc Completed in 0ms
npm timing config:load:user Completed in 0ms
npm timing config:load:file:C:\Users\juan_pablo_jofre\AppData\Roaming\npm\etc\npmrc Completed in 0ms
npm timing config:load:global Completed in 1ms
npm timing config:load:validate Completed in 0ms
npm timing config:load:credentials Completed in 0ms
npm timing config:load:setEnvs Completed in 1ms
npm timing config:load Completed in 10ms
npm timing npm:load:configload Completed in 10ms
npm timing npm:load:mkdirpcache Completed in 1ms
npm timing npm:load:mkdirplogs Completed in 0ms
npm verb title npm
npm verb argv "--loglevel" "verbose" "--version"
npm timing npm:load:setTitle Completed in 1ms
npm timing config:load:flatten Completed in 2ms
npm timing npm:load:display Completed in 12ms
npm verb logfile logs-max:10 dir:C:\Users\juan_pablo_jofre\AppData\Local\npm-cache\_logs
npm verb logfile C:\Users\juan_pablo_jofre\AppData\Local\npm-cache\_logs\2023-04-06T05_07_53_607Z-debug-0.log
npm timing npm:load:logFile Completed in 10ms
npm timing npm:load:timers Completed in 0ms
npm timing npm:load:configScope Completed in 0ms
npm timing npm:load Completed in 38ms
8.19.4
npm verb exit 0
npm timing npm Completed in 42ms
npm info ok
```

## System Info
```
OS Name	Microsoft Windows 10 Enterprise
Version	10.0.19042 Build 19042
Other OS Description 	Not Available
OS Manufacturer	Microsoft Corporation
System Name	***************
System Manufacturer	Dell Inc.
System Model	Precision 5560
System Type	x64-based PC
System SKU	0A62
Processor	11th Gen Intel(R) Core(TM) i7-11850H @ 2.50GHz, 2496 Mhz, 8 Core(s), 16 Logical Processor(s)
BIOS Version/Date	Dell Inc. 1.9.0, 3/17/2022
SMBIOS Version	3.2
Embedded Controller Version	255.255
BIOS Mode	UEFI
BaseBoard Manufacturer	Dell Inc.
BaseBoard Product	0WPMMN
BaseBoard Version	A00
Platform Role	Mobile
Secure Boot State	On
PCR7 Configuration	Binding Possible
Windows Directory	C:\WINDOWS
System Directory	C:\WINDOWS\system32
Boot Device	\Device\HarddiskVolume1
Locale	United States
Hardware Abstraction Layer	Version = "10.0.19041.2728"
User Name	***************
Time Zone	***************
Installed Physical Memory (RAM)	32.0 GB
Total Physical Memory	31.7 GB
Available Physical Memory	11.3 GB
Total Virtual Memory	63.7 GB
Available Virtual Memory	39.8 GB
Page File Space	32.0 GB
Page File	C:\pagefile.sys
```

## Node.js
- Installation folder: `C:\nodejs` 
- Behavior didn't change when installed in `C:\Program Files\nodejs`
- Behavior didn't change between versions 16 and 18 (either on `c:\nodejs` or `C:\Program Files\nodejs`)
- After each installation and test, the version of node.js was uninstalled from **Control Panel**.**Programs and Features**.**Uninstall or chagne a program**, and the machine rebooted.
- `nvm` (for Windows) is not installed, neither used to do the installs (and yes, I do know about `nvm` but I'm not using it in this project).  

## User 
- non admin user
- to install node (each time) I had to request privileged permissions to company IT; by default, all company users are non-admin.

**Comment by StefanStojanovic** [comment]
Can you run `where npm` and let me know the output? That will probably give you the path to a cmd file, can you post its content here as well?

How did you install Node.js? Can you check if it's in the path? If it's not, then `npm.cmd` must be in the same location as the node executable.

**Comment by rajatchauhan012** [comment]
![Screenshot 2024-08-16 013140](https://github.com/user-attachments/assets/cb308ac9-2646-47e9-b67c-f3560498fde4)
![Screenshot 2024-08-16 013155](https://github.com/user-attachments/assets/064aa2ba-6ad8-48ee-a330-2a6f422e29ad)
I am facing the same issue could you please help me???
